extension/README.md

```diff
@@ -0,0 +1,36 @@
+# Extensions
+
+Trillian defines a number of extension points to allow for customization by
+forks. At runtime, implementations are acquired via an [extension.Registry](
+https://github.com/google/trillian/blob/master/extension/registry.go#L10), which
```

Comment:
Maybe drop line numbers (here and elsewhere) and just link to the file?  The line numbers are going to go stale as time goes on.